src/api/mod.rs

```diff
@@ -3296,10 +3296,219 @@ mod test {
     );
   }
 
-  #[interpolate_test(8, 8)]
-  #[interpolate_test(10, 10)]
-  #[interpolate_test(16, 16)]
-  fn lookahead_size_properly_bounded(rdo_lookahead: u64) {
+  #[derive(Debug, Clone, Copy)]
+  struct LookaheadTestExpectations {
+    pre_receive_frame_q_len: usize,
+    pre_receive_fi_len: usize,
+    post_receive_frame_q_len: usize,
+    post_receive_fi_len: usize,
+  }
+
+  #[test]
+  fn lookahead_size_properly_bounded_8() {
+    const LOOKAHEAD_SIZE: u64 = 8;
+    const EXPECTATIONS: [LookaheadTestExpectations; 60] = [
+      LookaheadTestExpectations { pre_receive_frame_q_len: 1, pre_receive_fi_len: 0, post_receive_frame_q_len: 1, post_receive_fi_len: 0 },
```

Comment:
Doing this in a structure-of-arrays fashion rather than array-of-structures would allow to not repeat the same long field name every line. Something like:
```rust
let pre_receive_frame_q_len = [1, 2, 3, 4, ...];
let pre_receive_fi_len = [0, 0, 0, 0, ...];
...
```
Those arrays could also be inside the structure if you prefer.